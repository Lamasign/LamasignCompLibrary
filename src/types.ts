import {
	AccordionVariantsArray,
	AccordionVariantsEnum,
	AccordionVariantsMap,
} from "./components/accordion/BAccordion.types"


import {
	BadgeColorsArray,
	BadgeColorsEnum,
	BadgeColorsMap,
	BadgeSizeEnum,
	BadgeSizesArray,
	BadgeSizesMap,
	BadgeVariantsArray,
	BadgeVariantsEnum,
	BadgeVariantsMap,
} from './components/Badge/BBadge.types'

import {
	ButtonSizesEnum,
	ButtonVariantsEnum,
	ButtonTagsEnum,
	ButtonTypesEnum,
	ButtonSizesArray,
	ButtonSizesMap,
	ButtonVariantsArray,
	ButtonVariantsMap,
	ButtonTagsArray,
	ButtonTypesArray,
} from './components/Button/BButton.types'

import {
	CheckboxVariantsEnum,
	CheckboxVariantsArray,
	CheckboxsVariantsMap,
} from './components/Checkbox/BCheckbox.types'

import {
	DropdownVariantsEnum,
	DropdownDirectionEnum,
	DropdownDirectionMap,
	DropdownVariantsArray,
	DropdownVariantsMap,
} from './components/Dropdown/BDropdown.types'

import {
	FIELD_STATUS_INJECTION_KEY,
	FieldStatusesArray,
	FieldStatusesEnum,
	FieldStatusesClassesMap,
	FieldStatusesTypes,
	FieldMessageStatusesClassesMap
} from './components/Field/BField.types'

import {
	InputFileStatusesArray,
	InputFileStatusesClassesMap,
	InputFileStatusesEnum
} from './components/FileInput/BFileInput.types'

import {
	InputTypesEnum,
	InputTypesArray,
	InputStatusesArray,
	InputStatusesClassesMap,
	InputStatusesEnum,
	InputStatusesTypes,
	InputTypes,
} from './components/Input/BInput.types'

import {
	LinkDisabledClassesMap,
	LinkTags,
	LinkTagsArray,
	LinkTagsEnum,
	LinkVariantsArray,
	LinkVariantsEnum,
	LinkVariantsMap,
} from './components/Link/BLink.types'

import {
	MessageVariantsArray,
	MessageVariantsEnum,
	MessageVariantsMap,
} from './components/Message/BMessage.types'

import {
	NumberInputStatusesArray,
	NumberInputStatusesClassesMap,
	NumberInputStatusesEnum,
	NumberInputStatusesTypes,
} from './components/NumberInput/BNumberInput.types'

import {
	SelectColorsEnum,
	SelectColorsArray,
	SelectColorsMap,
} from './components/Select/BSelect.types'

import {
	SkeletonVariantsEnum,
	SkeletonVariantsArray,
	SkeletonVariantsMap,
	SkeletonAnimationsArray,
	SkeletonAnimationsEnum,
	SkeletonAnimationsMap,
	SkeletonColorsArray,
	SkeletonColorsEnum,
	SkeletonColorsMap,
} from './components/Skeleton/BSkeleton.types'

import {
	SwitchVariantsArray,
	SwitchVariantsEnum,
	SwitchVariantsMap,
} from './components/Switch/BSwitch.types'

import {
	TableColumnHeaderVariantsMap,
	TableColumnVariantsArray,
	TableColumnVariantsEnum,
	TableColumnVariantsMap,
} from './components/Table/BTable.types'

import {
	TagVariantsArray,
	TagVariantsEnum,
	TagVariantsMap,
	TagsVariantsMap,
} from './components/Tag/BTag.types'

import {
	ToastVariantsArray,
	ToastVariantsMap,
	ToastVariantsEnum,
} from './components/Toast/BToast.types'

import {
	TooltipPositionEnum,
	TooltipPositionsArray,
	TooltipPositionsMap,
	TooltipVariantsArray,
	TooltipVariantsEnum,
	TooltipVariantsMap,
} from './components/Tooltip/BTooltip.types'

export {
	AccordionVariantsArray,
	AccordionVariantsEnum,
	AccordionVariantsMap,
	BadgeColorsArray,
	BadgeColorsEnum,
	BadgeColorsMap,
	BadgeSizeEnum,
	BadgeSizesArray,
	BadgeSizesMap,
	BadgeVariantsArray,
	BadgeVariantsEnum,
	BadgeVariantsMap,
	ButtonSizesEnum,
	ButtonVariantsEnum,
	ButtonTagsEnum,
	ButtonTypesEnum,
	ButtonSizesArray,
	ButtonSizesMap,
	ButtonVariantsArray,
	ButtonVariantsMap,
	ButtonTagsArray,
	ButtonTypesArray,
	CheckboxVariantsEnum,
	CheckboxVariantsArray,
	CheckboxsVariantsMap,
	DropdownVariantsEnum,
	DropdownDirectionEnum,
	DropdownDirectionMap,
	DropdownVariantsArray,
	DropdownVariantsMap,
	FIELD_STATUS_INJECTION_KEY,
	FieldStatusesArray,
	FieldStatusesEnum,
	FieldStatusesClassesMap,
	FieldStatusesTypes,
	FieldMessageStatusesClassesMap,
	InputFileStatusesArray,
	InputFileStatusesClassesMap,
	InputFileStatusesEnum,
	InputTypesEnum,
	InputTypesArray,
	InputStatusesArray,
	InputStatusesClassesMap,
	InputStatusesEnum,
	InputStatusesTypes,
	InputTypes,
	LinkDisabledClassesMap,
	LinkTags,
	LinkTagsArray,
	LinkTagsEnum,
	LinkVariantsArray,
	LinkVariantsEnum,
	LinkVariantsMap,
	MessageVariantsArray,
	MessageVariantsEnum,
	MessageVariantsMap,
	NumberInputStatusesArray,
	NumberInputStatusesClassesMap,
	NumberInputStatusesEnum,
	NumberInputStatusesTypes,
	SelectColorsEnum,
	SelectColorsArray,
	SelectColorsMap,
	SkeletonVariantsEnum,
	SkeletonVariantsArray,
	SkeletonVariantsMap,
	SkeletonAnimationsArray,
	SkeletonAnimationsEnum,
	SkeletonAnimationsMap,
	SkeletonColorsArray,
	SkeletonColorsEnum,
	SkeletonColorsMap,
	SwitchVariantsArray,
	SwitchVariantsEnum,
	SwitchVariantsMap,
	TableColumnHeaderVariantsMap,
	TableColumnVariantsArray,
	TableColumnVariantsEnum,
	TableColumnVariantsMap,
	TagVariantsArray,
	TagVariantsEnum,
	TagVariantsMap,
	TagsVariantsMap,
	ToastVariantsArray,
	ToastVariantsMap,
	ToastVariantsEnum,
	TooltipPositionEnum,
	TooltipPositionsArray,
	TooltipPositionsMap,
	TooltipVariantsArray,
	TooltipVariantsEnum,
	TooltipVariantsMap,
}
