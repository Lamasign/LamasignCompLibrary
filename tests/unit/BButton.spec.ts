import { ButtonSizesEnum, ButtonTypesArray, ButtonVariantsArray } from './../../src/components/Button/BButton.types'
import BButton from '@/components/Button/BButton.vue'
import { mount } from "@vue/test-utils"
import { createWrapper } from './utils'

describe('BButton.vue', () => {
	it('render the content of his default slot', () => {
		expect.assertions(1)
		const slotContent = 'button'
		const wrapper = mount(BButton, {
			slots: { default: slotContent }
		})
		expect(wrapper.text()).toMatch(slotContent)
	})

	it('render the appropritate classe based on the "variant" prop', () => {
		expect.assertions(ButtonVariantsArray.length + 1)
		const defaultWrapper = createWrapper(BButton)

		expect(defaultWrapper.element).toMatchSnapshot()

		ButtonVariantsArray.forEach(variant => {
			const wrapperWithProps = createWrapper(BButton, {
				propsData: {
					variant
				}
			})
			expect(wrapperWithProps.element).toMatchSnapshot()
		})
	})

	it('renders right html type based on props called "type"', () => {
		expect.assertions(ButtonTypesArray.length + 1)
		const defaultWrapper = createWrapper(BButton)
		expect(defaultWrapper.find('button').attributes().type).toEqual('button')
		ButtonTypesArray.forEach(type => {
			const wrapperWithProps = createWrapper(BButton, {
				propsData: {
					type
				}
			})
			expect(wrapperWithProps.find('button').attributes().type).toEqual(type)
		})
	})

	it('renders with the appropriate classes based on the "outlined" prop', async () => {
		expect.assertions(ButtonVariantsArray.length * 2 + 2)

		const defaultWrapper = createWrapper(BButton)
		expect(defaultWrapper.element).toMatchSnapshot()

		await defaultWrapper.setProps({ outlined: true })
		expect(defaultWrapper.element).toMatchSnapshot()

		await Promise.all(ButtonVariantsArray.map(async (variant) => {
			const wrapperWithProps = createWrapper(BButton, {
				propsData: {
					variant
				}
			})
			expect(wrapperWithProps.element).toMatchSnapshot()
			await wrapperWithProps.setProps({ outlined: true })
			return Promise.resolve(expect(wrapperWithProps.element).toMatchSnapshot())
		}))
	})

	it('renders with the appropriate classes based on the "size" prop', async () => {
		const defaultWrapper = createWrapper(BButton)
		expect(defaultWrapper.element).toMatchSnapshot()

		await defaultWrapper.setProps({ size: ButtonSizesEnum.SMALL })
		expect(defaultWrapper.element).toMatchSnapshot()
	})

	it('renders with the appropriate classses based on the "disabled" prop', async () => {
		const defaultWrapper = createWrapper(BButton)
		expect(defaultWrapper.element).toMatchSnapshot()

		await defaultWrapper.setProps({ disabled: true })
		expect(defaultWrapper.element).toMatchSnapshot()

		// Disabled with no Outline
		await Promise.all(ButtonVariantsArray.map(async (variant) => {
			const wrapperWithProps = createWrapper(BButton, {
				propsData: {
					variant
				}
			})
			expect(wrapperWithProps.element).toMatchSnapshot()
			await wrapperWithProps.setProps({ disabled: true })
			return Promise.resolve(expect(wrapperWithProps.element).toMatchSnapshot())
		}))

		// Disabled with Outline
		await Promise.all(ButtonVariantsArray.map(async (variant) => {
			const wrapperWithProps = createWrapper(BButton, {
				propsData: {
					variant
				}
			})
			expect(wrapperWithProps.element).toMatchSnapshot()
			await wrapperWithProps.setProps({ disabled: true, outlined: true })
			return Promise.resolve(expect(wrapperWithProps.element).toMatchSnapshot())
		}))
	})

	it('emits a click "event" when the button is clicked', async () => {
		const wrapper = createWrapper(BButton)
		await wrapper.find('button').trigger('click')
		expect(wrapper.emitted().click).toBeTruthy()
	})
})