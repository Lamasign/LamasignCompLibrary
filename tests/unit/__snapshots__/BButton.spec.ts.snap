// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BButton.vue render the appropritate classe based on the "variant" prop 1`] = `
<button
  class="bg-blue text-white font-normal hover:bg-blue-dark focus:ring-4 ring-blue title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue render the appropritate classe based on the "variant" prop 2`] = `
<button
  class="bg-blue text-white font-normal hover:bg-blue-dark focus:ring-4 ring-blue title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue render the appropritate classe based on the "variant" prop 3`] = `
<button
  class="bg-orange text-white hover:bg-orange-dark focus:ring-4 ring-orange title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue render the appropritate classe based on the "variant" prop 4`] = `
<button
  class="bg-red text-black hover:text-white hover:bg-red-light focus:ring-4 ring-red title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue render the appropritate classe based on the "variant" prop 5`] = `
<button
  class="bg-green text-white hover:bg-green-dark focus:ring-4 ring-green title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue render the appropritate classe based on the "variant" prop 6`] = `
<button
  class="bg-purple text-white hover:text-white hover:bg-purple-dark focus:ring-4 ring-purple title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 1`] = `
<button
  class="bg-blue text-white font-normal hover:bg-blue-dark focus:ring-4 ring-blue title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 2`] = `
<button
  class="bg-white text-blue border border-blue hover:border-blue-dark hover:text-blue-dark focus:ring-4 ring-blue title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 3`] = `
<button
  class="bg-blue text-white font-normal hover:bg-blue-dark focus:ring-4 ring-blue title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 4`] = `
<button
  class="bg-orange text-white hover:bg-orange-dark focus:ring-4 ring-orange title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 5`] = `
<button
  class="bg-red text-black hover:text-white hover:bg-red-light focus:ring-4 ring-red title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 6`] = `
<button
  class="bg-green text-white hover:bg-green-dark focus:ring-4 ring-green title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 7`] = `
<button
  class="bg-purple text-white hover:text-white hover:bg-purple-dark focus:ring-4 ring-purple title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 8`] = `
<button
  class="bg-white text-blue border border-blue hover:border-blue-dark hover:text-blue-dark focus:ring-4 ring-blue title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 9`] = `
<button
  class="bg-white text-orange border border-orange hover:border-orange-dark hover:text-orange-dark focus:ring-4 ring-orange title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 10`] = `
<button
  class="bg-white text-red border border-red hover:border-red-light hover:text-red-light focus:ring-4 ring-red title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 11`] = `
<button
  class="bg-white text-green border border-green hover:border-green-dark hover:text-green-dark focus:ring-4 ring-green title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "outlined" prop 12`] = `
<button
  class="bg-white text-purple border border-purple hover:border-purple-dark hover:text-purple-dark focus:ring-4 ring-purple title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "size" prop 1`] = `
<button
  class="bg-blue text-white font-normal hover:bg-blue-dark focus:ring-4 ring-blue title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classes based on the "size" prop 2`] = `
<button
  class="bg-blue text-white font-normal hover:bg-blue-dark focus:ring-4 ring-blue title-4 p-2 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 1`] = `
<button
  class="bg-blue text-white font-normal hover:bg-blue-dark focus:ring-4 ring-blue title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 2`] = `
<button
  aria-disabled="true"
  class="bg-blue/50 text-white title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 3`] = `
<button
  class="bg-blue text-white font-normal hover:bg-blue-dark focus:ring-4 ring-blue title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 4`] = `
<button
  class="bg-orange text-white hover:bg-orange-dark focus:ring-4 ring-orange title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 5`] = `
<button
  class="bg-red text-black hover:text-white hover:bg-red-light focus:ring-4 ring-red title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 6`] = `
<button
  class="bg-green text-white hover:bg-green-dark focus:ring-4 ring-green title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 7`] = `
<button
  class="bg-purple text-white hover:text-white hover:bg-purple-dark focus:ring-4 ring-purple title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 8`] = `
<button
  aria-disabled="true"
  class="bg-blue/50 text-white title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 9`] = `
<button
  aria-disabled="true"
  class="bg-orange/50 text-white title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 10`] = `
<button
  aria-disabled="true"
  class="bg-red/50 text-white title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 11`] = `
<button
  aria-disabled="true"
  class="bg-green/50 text-white title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 12`] = `
<button
  aria-disabled="true"
  class="bg-purple/50 text-white title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 13`] = `
<button
  class="bg-blue text-white font-normal hover:bg-blue-dark focus:ring-4 ring-blue title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 14`] = `
<button
  class="bg-orange text-white hover:bg-orange-dark focus:ring-4 ring-orange title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 15`] = `
<button
  class="bg-red text-black hover:text-white hover:bg-red-light focus:ring-4 ring-red title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 16`] = `
<button
  class="bg-green text-white hover:bg-green-dark focus:ring-4 ring-green title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 17`] = `
<button
  class="bg-purple text-white hover:text-white hover:bg-purple-dark focus:ring-4 ring-purple title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 18`] = `
<button
  aria-disabled="true"
  class="bg-white text-blue/50 border border-blue/50 title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 19`] = `
<button
  aria-disabled="true"
  class="bg-white text-orange/50 border border-orange/50 title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 20`] = `
<button
  aria-disabled="true"
  class="bg-white text-red/50 border border-red/50 title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 21`] = `
<button
  aria-disabled="true"
  class="bg-white text-green/50 border border-green/50 title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;

exports[`BButton.vue renders with the appropriate classses based on the "disabled" prop 22`] = `
<button
  aria-disabled="true"
  class="bg-white text-purple/50 border border-purple/50 title-3 p-3 uppercase rounded cursor-pointer disabled:cursor-not-allowed"
  disabled="disabled"
  type="button"
/>
`;
